proposals/interface-types/working-notes/choices/webidl.md

```diff
@@ -0,0 +1,228 @@
+float` in WebIDL is one that is not common in programming languages.
+
+Similarly,  WebIDL  has   three  variants  of  string   type:  the  `DOMString`,
+`ByteString` and `USVString` string. Again, the distinction between `DOMString`s
+and `USVString`s is not one that is common in programming languages.
+
+### Extended Attributes
+
+Heavy use is made in WebIDL of so-called extended attributes. For example, the
+`AllowShared` extended attribute is intended to mark a buffer type to be backed
+by a `SharedArrayBuffer`.
+
+Apart from the obvious dependence on a specific feature of JavaScript (shared
+array buffers), allowing sharing on values contradicts one of the key design
+criteria for Interface Types -- namely supporting interoperability in a cross
+ownership domain where there is no sharing.
+
+In practice, Web APIs that are specified using WebIDL make heavy use of extended
+attributes to convey additional semantics that are very JavaScript centric. Such
+usages would have to be prohibited when designing APIs for languages other than
+JavaScript.
+
+This again leads to a situation where the 'real' IDL that would be usable for
+WebAssembly would be a limited form of a public IDL. This is not an ideal model
+for standardization nor for supporting the wider WebAssembly ecosystem.
+
+### Nominal Types
+
+JavaScript does not have a static model of types. On the other hand, many
+programming languages do have static types. One of the features of many static type systems is the so-called _nominal type_.
+
+A nominal type differs from a structural type in several ways; but one of the
+most fundamental ways is that a nominal type can be used to model entities that
+are not directly conveyed by the actual data used to denote them.
+
+For example, one may choose to model a chair using the nominal type `Chair`;
+whose values contain a pair of integers: the SKU and price (for example). The
+pair of integers is _not equivalent_ to a chair, but represent the information
+needed to model chairs in some application or API. 
+
+A different entity (a person say) may also be modeled by a type `Person` which
+contains two integers (for example, an indentification number and an index into
+a street address table).
+
+A given pair of integers might be either a `Chair` or a `Person` (or a 2D point)
+and there is no way of conveying the distinction unless the nominal type is also
+taken into consideration.
+
+### Algebraic Type Definitions
+
+The Interface Types language permits the definition of a limited form of
+_algebraic type definition_. This is, in part, to regularize concepts such as
+nullability and to permit limited forms of polymorphism (allowing a color to be
+named by a string, a vector of small integers or a large integer, for example).
+
+WebIDL explicitly prohibits the formation of new types; indeed it uses type
+unions to achieve similar polymorphism. However, as we note above, not all
```

Comment:
In a sense, Web IDL does allow the formation of new types (`dicationary`, `interface`, ...).  Even Web IDL's `union` is basically a type constructor.  Trying to say it doesn't will lead to arguments over subtle definitions which are probably best avoided.

Reply:
There is an important constraint here: JS supports variants because a program can inspect the data and discover which variant is actually in use. This is not possible without labeled sum types; which webIDL does not have. Not that willing to drop this.

Reply:
removed.